NorthstarDLL/mods/modmanager.cpp

```diff
@@ -161,6 +161,91 @@ Mod::Mod(fs::path modDir, char* jsonBuf)
 		}
 	}
 
+	if (modJson.HasMember("ConCommands") && modJson["ConCommands"].IsArray())
+	{
+		for (auto& concommandObj : modJson["ConCommands"].GetArray())
+		{
+			if (!concommandObj.IsObject() || !concommandObj.HasMember("Name") || !concommandObj.HasMember("Function") ||
+				!concommandObj.HasMember("Context"))
+			{
+				continue;
+			}
+
+			// have to allocate this manually, otherwise concommand registration will break
+			// unfortunately this causes us to leak memory on reload, unsure of a way around this rn
+			ModConCommand* concommand = new ModConCommand;
+			concommand->Name = concommandObj["Name"].GetString();
+			concommand->Function = concommandObj["Function"].GetString();
+			concommand->Context = ScriptContextFromString(concommandObj["Context"].GetString());
+			if (concommand->Context == ScriptContext::INVALID)
+			{
+				spdlog::warn("Mod ConCommand {} has invalid context {}", concommand->Name, concommandObj["Context"].GetString());
+				continue;
+			}
+
+			if (concommandObj.HasMember("HelpString"))
+				concommand->HelpString = concommandObj["HelpString"].GetString();
+			else
+				concommand->HelpString = "";
+
+			concommand->Flags = FCVAR_NONE;
+
+			if (concommandObj.HasMember("Flags"))
+			{
+				// read raw integer flags
+				if (concommandObj["Flags"].IsInt())
+				{
+					concommand->Flags = concommandObj["Flags"].GetInt();
+				}
```

Comment:
the string flags parsing code here is duplicated from convars, it'd be nice if it was all in one helper function instead